0-intro-to-programming/slides-code/0-intro-to-programming.Rmd

```diff
@@ -0,0 +1,720 @@
+  ```
+]
+
+---
+class: middle
+
+.pull-left[
+
+## Repeating parts of the code
+
+When using **loops** we need a way to decide how many iterations in a loop.
+
+* **Hard coded**. Repeat an exact number of times. For example, 10 times
+
+* **While-loop**. Repeat until a condition is met. For example, a match is found
+
+* **Dynamically**. Use a characteristic in the data 
+to decide the number of repeats. 
+For example, repeat once for each country in the data set.
+
+]
+
+.pull-right[
+  ```{r, eval=TRUE, echo=FALSE,out.width = '75%'}
+    knitr::include_graphics("img/code-flow-loop.png")
+  ```
+]
+---
+class: middle
+
+.pull-left[
+
+## Sometimes skipping parts of the code
+
+When repeating or re-using code, 
+some sections does not always apply.
+We use **if-conditions** to instruct the computer
+when to skip some code. 
+
+* Only execute certain blocks on certain countries in a loop
+
+* Code intended to be re-used across projects can fit more cases
+when **if-conditions** are used
+to make the code adapt itself to what is needed in each context.
+
+]
+
+.pull-right[
+  ```{r, eval=TRUE, echo=FALSE,out.width = '75%'}
+    knitr::include_graphics("img/code-flow-if.png")
+  ```
+]
+
+---
+class: middle
+
+.pull-left[
+
+  ## Code flow - combine it all
+  
+  Often loops and if-conditions are combined or they can be nested. 
+  
+  Session 2 of this course will show how this is done in Python. 
+  
+  Having neat, human readable and intuitive code flow 
+  using loops and if-conditions,
+  is important to make code _good code_.
+]
+
+.pull-right[
+  ```{r, eval=TRUE, echo=FALSE,out.width = '75%'}
+    knitr::include_graphics("img/code-flow-example.png")
+  ```
+] 
+
+---
+class: middle
+
+## Version control
+
+.pull-left[
+
+  _"Version control"_ refers to tools used to:
+
+  * Keep track of multiple people 
+  making different contributions 
+  to the files of a project
+  * Give access to previous versions
+  of the files of a project
+  
+  
+  Version control exists outside programming. 
+  For example, `ctrl+z`, 
+  restore files in OneDrive or Dropbox,
+  recovered files in Windows, etc.
+  
+  In coding version control is much more comprehensive,
+  and can track every single edit anyone in the team do.
+
+]
+
+.pull-left[
+  .center[
+    ```{r, eval=TRUE, echo=FALSE,out.width = '90%'}
+      knitr::include_graphics("img/commit.png")
+    ```
+  ]
+] 
+
+
+---
+class: middle
+
+## Version control of code = Version control of all data work
+  
+  Code is very exact instructions on 
+  how to create outputs from original data.
+  
+  If we handle the original data with the same level of exactness
+  (backed-up or properly cited), 
+  and if we version control our code, 
+  then all outputs are also version controlled.
+  
+  Version controlling all work is perhaps
+  the most valuable feature of programming.
+  
+  That standard for version control in code 
+  is called _Git_ or _GitHub_.
```

Comment:
I wouldn't call git a standard, and definitely not github. Version control is much bigger than git. There's cvs and svn and others besides git. Git is just the most popular version control software and github is one of the many git repository hosting services alongside GitLab, Bitbucket, GitBucket etc.

Reply:
In 3c90caab91c7d1e811a265979d970722c4ba2829 I am softening the language to

>   The most common standard for version control in code is called Git and most Git projects are hosted on GitHub.

Sounds good?